### DISCUSSION TITLE
Script for single rooms

### DISCUSSION THREAD

**Discussion by Rebislori** [Tasshack/dreame-vacuum#116] [Q&A] [answered]
Hi everyone again;
I created some scripts (one for each room) to voice control via alexa my dreame vacuum.
I'd like one script only for vacuum, and one for vscuum/mop, but it looks like "water_volume" is not the way to go.
I use the "vscuum_clean_segment" service, and disabled "custom cleaning" in mi home app.

Thanks to everyone!

**Comment by Tasshack** [comment, marked as answer]
You can edit mop/vacuum mode for each room from customized cleaning settings like this https://github.com/Tasshack/dreame-vacuum/issues/44

**Comment by Rebislori** [comment]
Thanks,
`service: vacuum.send_command
data:
  command: action
  params:
    did: "6.2"
    siid: 6
    aiid: 2
    in:
      - piid: 4
        value: "{\"customeClean\":[[4,2,2,2,0],[5,0,2,3,1],[3,3,2,2,1],[2,2,4,1,0],[1,3,2,2,2]]}"
target:
  entity_id: vacuum.vacuum 
`
Apart for entity_id and values in square parentesis, did, siiid, aiiid, remain always the same for this command?

**Comment by Tasshack** [comment]
You need to edit values in the square parentheses (custom cleaning settings for each room) and to send all room settings at the same time. You cannot edit only single room with this method.

**Reply by Rebislori** [reply]
oh, got it.

So the script should include settings for all the rooms and then the command to clean the single room?

**Reply by Tasshack** [reply]
If you want customized cleaning to be stay on answer is yes. Alternatively you could turn off the customized cleaning and set the cleaning mode before starting the room cleaning like you previously suggested. Robot ignores current cleaning mode, water level and suction settings when customized cleaning is enabled thats why those settings are unavailable when customized cleaning is enabled.

**Reply by Rebislori** [reply]
Thanks!

**Comment by TRusselo** [comment]
@Rebislori - can you share your "clean single room" script?
im trying to figure that out, but my attempts do not work.

**Reply by TRusselo** [reply]
Found it
```
service: dreame_vacuum.vacuum_clean_segment
data:
  suction_level: 4
  segments: 6
  repeats: 1
  water_volume: 1
target:
  entity_id: vacuum.bitch_vac
```

**Reply by Rebislori** [reply]
`alias: "Gian: aspira il bagno della linda"
sequence:
  - service: switch.turn_off
    data: {}
    target:
      entity_id: switch.gian_robot_d_acciaio_customized_cleaning
  - service: select.select_option
    data:
      option: sweeping
    target:
      entity_id: select.gian_robot_d_acciaio_cleaning_mode
  - service: dreame_vacuum.vacuum_clean_segment
    data:
      suction_level: 3
      segments: 7
    target:
      entity_id: vacuum.gian_robot_d_acciaio
mode: single
icon: mdi:fridge
`
I switch off customized cleaning before starting 💪

**Comment by Tasshack** [comment]
https://github.com/Tasshack/dreame-vacuum/releases/tag/v2.0.0b0
